src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java

```diff
@@ -0,0 +1,255 @@
+        removeAndRestoreTypicalDataInDatastore();
+        
+        String fromEmail = "Admin@" + Config.getAppId() + ".appspotmail.com";
+        String fromName = "TEAMMATES Admin";
+        replyTo = "teammates@comp.nus.edu.sg";
+        InternetAddress internetAddress = new InternetAddress(fromEmail, fromName);
+        from = internetAddress.toString();
+    }
+    
+    @Test
+    public void testGenerateFeedbackSessionEmails() throws Exception {
+        FeedbackSessionAttributes session = fsLogic.getFeedbackSession("First feedback session", "idOfTypicalCourse1");
+        
+        CourseAttributes course = coursesLogic.getCourse(session.getCourseId());
+        
+        List<StudentAttributes> students = studentsLogic.getStudentsForCourse(session.getCourseId());
+        List<InstructorAttributes> instructors = instructorsLogic.getInstructorsForCourse(session.getCourseId());
+        
+        StudentAttributes student1 = studentsLogic.getStudentForEmail(course.getId(), "student5InCourse1@gmail.tmt");
+        StudentAttributes student2 = studentsLogic.getStudentForEmail(course.getId(), "student2InCourse1@gmail.tmt");
+        StudentAttributes student3 = studentsLogic.getStudentForEmail(course.getId(), "student3InCourse1@gmail.tmt");
+        
+        InstructorAttributes instructor1 =
+                instructorsLogic.getInstructorForEmail(course.getId(), "instructor1@course1.tmt");
+        InstructorAttributes instructor2 =
+                instructorsLogic.getInstructorForEmail(course.getId(), "instructor2@course1.tmt");
+        
+        ______TS("feedback session opening emails");
+        
+        List<MimeMessage> emails = new EmailGenerator().generateFeedbackSessionOpeningEmails(session);
+        assertEquals(10, emails.size());
+        
+        String subject = String.format(EmailType.FEEDBACK_OPENING.getSubject(),
+                                       course.getName(), session.getFeedbackSessionName());
+        
+        String submitUrl = Config.getAppUrl(Const.ActionURIs.STUDENT_FEEDBACK_SUBMISSION_EDIT_PAGE)
+                                 .withCourseId(course.getId())
+                                 .withSessionName(session.getFeedbackSessionName())
+                                 .withRegistrationKey(StringHelper.encrypt(student1.key))
+                                 .withStudentEmail(student1.email)
+                                 .toAbsoluteString();
+        
+        String textInEmail = "Hello " + student1.name + "{*}The following feedback session is now open"
+                             + "{*}" + course.getId() + "{*}" + course.getName()
+                             + "{*}" + session.getFeedbackSessionName()
+                             + "{*}" + submitUrl + "{*}" + submitUrl;
+        
+        verifyEmail(emails.get(0), student1.email, subject, textInEmail);
+        
+        verifyEmail(emails.get(1), student2.email, subject, "The following feedback session is now open");
```

Comment:
The instructor should receive a copy of the email informing him/her that the emails to the students have been sent out. I think the old tests don't test it, but perhaps we should test it at some point